includes/Modules/Analytics/Advanced_Tracking.php

```diff
@@ -0,0 +1,204 @@
+			return;
+		}
+		$this->configure_events();
+		// TODO: Instantiate and register Metadata_Collector here.
+		( new Measurement_Code_Injector( $this->event_configurations ) )->inject_event_tracking();
+	}
+
+	/**
+	 * Adds triggers to AMP configuration.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @param array $gtag_amp_opt gtag config options for AMP.
+	 * @return array $gtag_amp_opt gtag config options for AMP.
+	 */
+	private function set_up_advanced_tracking_amp( $gtag_amp_opt ) {
+		$this->configure_events();
+
+		if ( ! array_key_exists( 'triggers', $gtag_amp_opt ) ) {
+			$gtag_amp_opt['triggers'] = array();
+		}
+		foreach ( $this->event_configurations as $event_config ) {
+			$gtag_amp_opt['triggers'][ $event_config->get_amp_trigger_name() ] = $event_config->to_amp_config();
+		}
+		return $gtag_amp_opt;
+	}
+
+	/**
+	 * Creates list of event configurations.
+	 *
+	 * @since n.e.x.t.
+	 */
+	private function configure_events() {
+		$this->active_plugins = $this->plugin_detector->determine_active_plugins( $this->get_supported_plugins() );
+
+		$this->event_configurations = array();
+		foreach ( $this->active_plugins as $plugin_config ) {
+			$measurement_event_list = $plugin_config['event_config_list'];
+			if ( null !== $measurement_event_list ) {
+				foreach ( $measurement_event_list->get_events() as $measurement_event ) {
+					$this->event_configurations[] = $measurement_event;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Returns list of event configurations.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @return array The list of event configurations.
+	 */
+	public function get_event_configurations() {
+		return $this->event_configurations;
+	}
+
+
+	/**
+	 * Returns list of supported plugins.
+	 *
+	 * @since n.e.x.t.
+	 *
+	 * @return array The list of supported plugins.
+	 */
+	public function get_supported_plugins() {
+		if ( null == $this->supported_plugins ) {
+			$this->supported_plugins = array(
+				'Contact Form 7'   => array(
+					'check_name'        => 'WPCF7_PLUGIN_DIR',
+					'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+					'event_config_list' => new CF7_Event_List(),
+				),
+				'Formidable Forms' => array(
+					'check_name'        => 'load_formidable_forms',
+					'check_type'        => Plugin_Detector::TYPE_FUNCTION,
+					'event_config_list' => new FormidableForms_Event_List(),
+				),
+				'Ninja Forms'      => array(
+					'check_name'        => 'NF_PLUGIN_DIR',
+					'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+					'event_config_list' => new NinjaForms_Event_List(),
+				),
+				'WooCommerce'      => array(
+					'check_name'        => 'WC_PLUGIN_FILE',
+					'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+					'event_config_list' => new Woocommerce_Event_List(),
+				),
+				'WPForms'          => array(
+					'check_name'        => 'WPFORMS_PLUGIN_DIR',
+					'check_type'        => Plugin_Detector::TYPE_CONSTANT,
+					'event_config_list' => new WPForms_Event_List(),
```

Comment:
As discussed earlier, let's provide class names for `event_config_list` instead of class instances, and then instantiate them in the `configure_events` method. This way only the classes for active plugins are instantiated.